packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -77,40 +79,34 @@ class Select extends Foundation<SelectHandledProps, SelectUnhandledProps, Select
     }
 
     public componentDidUpdate(prevProps: SelectProps): void {
-        let updatedSelection: ListboxItemProps[];
+        let shouldUpdateSelection: boolean = false;
         let updatedMenuVisibility: boolean = this.state.isMenuOpen;
 
-        if (prevProps.selectedItems !== this.props.selectedItems) {
-            updatedSelection = Listbox.getListboxItemDataFromIds(
-                this.props.selectedItems,
-                this.props.children
-            );
-        }
-
         if (prevProps.multiselectable !== this.props.multiselectable) {
-            // in the case that the multiselect is turned off but there are multiple items selected,
-            // choose the first item in the list of selected items
-            if (
-                this.props.multiselectable === false &&
-                this.state.selectedItems.length > 1
-            ) {
-                updatedSelection = [this.state.selectedItems[0]];
-            }
-
+            shouldUpdateSelection = true;
             updatedMenuVisibility = this.checkPropsForMenuState();
         }
 
         if (prevProps.isMenuOpen !== this.props.isMenuOpen) {
             updatedMenuVisibility = this.checkPropsForMenuState();
         }
 
-        if (updatedSelection !== undefined) {
-            this.updateSelection(updatedSelection);
-        }
-
         if (updatedMenuVisibility !== this.state.isMenuOpen) {
             this.toggleMenu(updatedMenuVisibility);
         }
+
+        if (prevProps.selectedItems !== this.props.selectedItems) {
```

Comment:
Should this return here? It kinda doesn't it make the effects of the previous checks moot?